lintball.sh

```diff
@@ -1,114 +1,130 @@
+  WORKING_DIR="${WORKING_DIR}/git"
+  FILENAMES="TODO"
+  git clone "${GIT_URI}" "${WORKING_DIR}"
+
+  # Checkout the branch we are interested in (in a subshell, so as not to affect PWD)
+  #shellcheck disable=SC2091
+  $(cd "${WORKING_DIR}" && git checkout "${GIT_COMMIT}" -B "${GIT_BRANCH}")
+  #printf  "running: git checkout \"${GIT_COMMIT}\" -B \"${GIT_BRANCH}\"\nOn workdir ${WORKING_DIR}\n"
+  FILENAMES=$(cd "${WORKING_DIR}" && git diff --name-only "${GIT_BRANCH}" "$(git merge-base "${GIT_BRANCH}" master)")
+  mapfile -t INPUT_FILES <<< "$FILENAMES"
 fi
 
-if ! [ -x "$(command -v shellcheck)" ]; then
-  echo 'Error: shellcheck not found' >&2
-  exit 1
-fi
 
-if ! [ -x "$(command -v pylint)" ]; then
-  echo 'Error: pylint not found' >&2
-  exit 1
+if [ -z ${FILENAMES+x} ]; then
+  echo "Error - filename(s) not passed to the script"
+  usage 1
 fi
 
-if ! [ -x "$(command -v cfn-lint)" ]; then
-  echo 'Error: cfn-lint not found' >&2
-  exit 1
-fi
+declare LINTIGNORE_PATH="${WORKING_DIR}/${LINTIGNORE_FILENAME}"
+echo "DEBUG = ${DEBUG}"
+#export DEBUG="false"
+debug "WORKING_DIR=${WORKING_DIR}"
+debug "LINTIGNORE_PATH=${LINTIGNORE_PATH}"
+debug "${WORKING_DIR} files: $(ls -alF ${WORKING_DIR})"
 
-if ! [ -x "$(command -v yamllint)" ]; then
-  echo 'Error: yamllint not found' >&2
-  exit 1
-fi
+RC=0
 
+declare PROCESS_FILE_FLAG=""
 
-# Assign "unique" filename to the results using timestamp
-OUTFILE="/scan/lintresults.$(date +%s%N)"
-# Ensure outfile is empty just in case it pre-exists
-true > "${OUTFILE}"
 
-FILENAME=/scan/$1
+for FILE in "${INPUT_FILES[@]}"
+do
+  PROCESS_FILE_FLAG="false"
+  FILENAME="${WORKING_DIR}/${FILE}"
 
-# Set an exit code
-RC=0
-# Confirm file exists
-if [ -f "${FILENAME}" ]
-then
-    if echo "${FILENAME}" | grep \.sh$  > /dev/null
-    then
-        #LD_LIBRARY_PATH is to get around an error running shellcheck on docker environments
-        echo "=========" >> "${OUTFILE}"
-        echo "Shellcheck running on ${FILENAME}" >> "${OUTFILE}"
-        # LD_LIBRARY_PATH=/tmp
-        shellcheck "${FILENAME}" >> "${OUTFILE}" 2>&1 || RC=1
-    fi
 
-    if echo "${FILENAME}" | grep -e \.py$ -e \.python$ > /dev/null
-    then
-        echo "=========" >> "${OUTFILE}"
-        echo "pylint running on ${OUTFILE}" >> "${OUTFILE}"
-
-        # Disable rule: E0401: Unable to import '<module>' (import-error)
-        # Reason:
-        #  - Lambdas are packaged in a separate process and uploaded to S3
-        #  - If we have this rule, we will have to run pip install on all lambdas, which is out of scope for this Script.
-        #
-        pylint "${FILENAME}" --disable E0401 >> "${OUTFILE}" 2>&1 || RC=1
+  if [ -f "${FILENAME}" ]
+  then
+    debug "$(ls -alF "${FILENAME}")"
+  fi
+
+  # Does the given .lintignore file exist
+  if [ -f "${LINTIGNORE_PATH}" ] ; then
+
+    debug "Found ${LINTIGNORE_FILENAME} file"
+    # Ignore if file is in lintignore
+    if grep -q "${FILE}" "${LINTIGNORE_PATH}"; then
```

Comment:
I'd suggest that this either needs to be "^${FILE}$" or we need to call out that the files in the lintignore file are effectively a regex